### DISCUSSION TITLE
page backgroundcolor modification with plugins

### DISCUSSION THREAD

**Discussion by SaElmas** [xournalpp/xournalpp#3006] [Q&A]
Is it possible to modify the page color by plugins.
It seams that non of the below commands are applicable for this.
Thanks.

app.msgbox for displaying messages to the user and letting the user choose an option by clicking a button
app.saveAs for retrieving the filename from a native saveAs dialog.
app.registerUi used in the initUi-function
app.uiAction for simulating a toolbar/menubar click starting some action,
app.uiActionSelected for notifying action listeners about selected options,
app.changeCurrentPageBackground for changing the background of the current page,
app.sidebarAction for actions accessible in the sidebar
app.layerAction for actions accessible in the layer controller
app.changeToolColor for changing the color of any tool with color capabilities,
app.changePdfBackgroundPageNr to change the pdf background page number
app.getDocumentStructure to get lots of useful info on the document, also used for applying operations on all/selected pages
app.scrollToPagescrolls relatively or absolutely to a page
app.scrollToPos scrolls relatively or absolutely to a new position on the same page
app.setCurrentPage sets the given page as new current page
app.setPageSize sets the width and height of the current page
app.setCurrentLayer sets the given layer as the new current layer, and updates visibility if specified
app.setLayerVisibility

**Comment by rolandlo** [comment]
As you correctly observed it is not possible with the current plugin API.
